homeassistant/components/radarr/coordinator.py

```diff
@@ -183,18 +184,22 @@ async def async_get_events(
 
     async def _async_get_events(self, _date: date) -> None:
         """Return events from specified date."""
+        info = await dt_util.async_get_time_zone(self.hass.config.time_zone)
+        offset = info and info.utcoffset(dt_util.now())
+        days = timedelta(days=0 if offset and offset < timedelta() else 1)
         self._events.extend(
-            _get_calendar_event(evt)
+            _get_calendar_event(days, evt)
             for evt in await self.api_client.async_get_calendar(
                 start_date=_date, end_date=_date + timedelta(days=1)
             )
             if evt.title not in (e.summary for e in self._events)
         )
 
 
-def _get_calendar_event(event: RadarrCalendarItem) -> RadarrEvent:
+def _get_calendar_event(offset: timedelta, event: RadarrCalendarItem) -> RadarrEvent:
     """Return a RadarrEvent from an API event."""
     _date, _type = event.releaseDateType()
+    _date += offset
```

Comment:
This looks weird. Cant you just construct a timezone aware date here instead?

Reply:
Agreed it does look weird even if it works. I just changed it a bit. Do you have a direct code change suggestion?

Reply:
I think you should be able to : _date.replace(tzinfo=info) with the timezone info you know the api returns in.

Reply:
That works for `datetime`. `_date` is a `date` object. Radarr gets the user time zone from the web browser. I do not see a time zone setting to get that information from the API.

Reply:
You can use the timezone in ha, to make the object aware?

Reply:
I still give this weird. Can you explain what you do in words? 

Why can't we just calculate the starting time and ending time in full date time instances. Then convert to timezone radarr expects and cut off the time parts.

Reply:
We use date objects to make the events all day events. The API from Radarr does not give the date and time with timezone information to my knowledge. As I mentioned, Radarr (I assume it's frontend) displays the events based on the timezone it sees from the web browser but we do not see that information at the API level.

Reply:
Sure. But you make the assumption that the timezone of the instance is the same as the frontend right, which is fine.

What im saying is don't make it a date only object until last instance. Stay as datetime all the way until you need to send it to the api.

Reply:
> Radarr gets the user time zone from the web browser. I do not see a time zone setting to get that information from the API.

Can you give some more details here? Are the dates returned from the Radarr API naive or timezone aware? If they are naive is there a way we can know what time zone they are anchored in?